sqlglot/dialects/snowflake.py

```diff
@@ -100,6 +101,14 @@ def _parse_date_part(self):
     return self.expression(exp.Extract, this=this, expression=expression)
 
 
+def _datatype_sql(self, expression):
+    if expression.this == exp.DataType.Type.ARRAY:
+        return "ARRAY"
```

Comment:
so these are needed because snowflake doesn't have these as nested data structures but using variants?

Reply:
Yep that is right